### DISCUSSION TITLE
MsQuic v1.5.0

### DISCUSSION THREAD

**Discussion by nibanks** [microsoft/msquic#1848] [Show and tell]
Supports both [version 1](https://datatracker.ietf.org/doc/html/rfc9000) and [draft-29](https://tools.ietf.org/html/draft-ietf-quic-transport-29) specification versions. Full details [here](https://github.com/microsoft/msquic/blob/main/docs/Release.md#msquic-v15-prerelease).

- Both Windows and Linux are officially supported.
- Both Schannel and (quictls fork of) OpenSSL TLS libraries are officially supported.
- MacOS still in "alpha" support.
- Unsigned artifacts available below.

**New Since v1.4**

- Windows Kernel client certificate support (#1652).
- Added load balancer app (#1696) and automated testing (#1707).
- Fixed a bug for QUIC_SETTING size validation logic (#1724).
- Added support for local UDP port sharing (Linux only) (#1751).
- Switched Param Id's to not be overlapping (#1758).
- Refactored library load (#1748) and added static linking support (Windows only) (#1446).
- Fixed bug around handshake idle timeout (#1780).
- Added support for stream prioritization (#1778).
- Fixed bug when shutting connection down during the handshake (#1797).
- Improved testing around path changes and few minor bug fixes (#1783, #1801).
- Improved event callback reentrancy to reduce app complexity (#1802).
- Fixed bug related to sharing abort code for stream send/recv shutdown (#1809).
- Fixed bug related to queuing 0-RTT during the handshake (but after start) (#1817).
- Use root level certificate verification callback in OpenSSL (#1818).
- Ignore Duplicate Stream Read Shutdowns (#1822).
- Constrain processor index to processor count on posix (#1824).
- Build speed and build dependency improvements (#1845) (#1839).
- Initial support for Android binaries (#1835).
- Update version negotiation to draft v4 (#1826).
- Fix issues where datapaths and sockets might be used incorrectly (#1843) (#1837).
- Various documentation improvements and additions.

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/msquic/releases/tag/v1.5.0'>MsQuic v1.5.0</a>.</em>

**Comment by wegylexy** [comment]
cmake ninja win-arm64 openssl won't build; same config worked for v1.4.0
```
ninja : error : '_deps/opensslquic-build/openssl/release/lib/libcrypto.lib', needed by 'bin/Release/msquic.dll', missing and no known rule to make it
```
